README.md

```diff
@@ -1,17 +1,192 @@
-## My Project
+# A Unified Efficient Pyramid Transformer for Semantic Segmentation
 
-TODO: Fill this README out!
+## Installation
 
-Be sure to:
+* Linux, CUDA>=10.0, GCC>=5.4
+* Python>=3.7
+* Create a conda environment:
 
-* Change the title in this README
-* Edit your repository description on GitHub
+```bash
+    conda create -n unept python=3.7 pip
+```
 
-## Security
+Then, activate the environment:
+```bash
+    conda activate unept
+```
+* PyTorch>=1.5.1, torchvision>=0.6.1 (following instructions [here](https://pytorch.org/))
+
+For example:
+```
+conda install pytorch==1.5.1 torchvision==0.6.1 cudatoolkit=10.2 -c pytorch
+```
+
+* Install [MMCV](https://mmcv.readthedocs.io/en/latest/), [MMSegmentation](https://github.com/open-mmlab/mmsegmentation/blob/master/docs/install.md), [timm](https://pypi.org/project/timm/)
+
+```
+pip install -r requirements.txt
+```
+
+* Install [Deformable DETR](https://github.com/fundamentalvision/Deformable-DETR) and compile the CUDA operators
+
+
+## Data Preparation
+Please following the code from [openseg](https://github.com/openseg-group/openseg.pytorch) to generate ground truth for boundary refinement. 
+
+The data format should be like this.
+
+### ADE20k
+
+```
+path/to/ADEChallengeData2016/
+  images/
+    training/
+    validation/
+  annotations/ 
+    training/
+    validation/
+  dt_offset/
+    training/
+    validation/
+```
+### PASCAL-Context
+```
+path/to/PASCAL-Context/
+  train/
+    image/
+    label/
+    dt_offset/
+  val/
+    image/
+    label/
+    dt_offset/
+```
+
+## Usage 
+### Training 
+**The default is for multi-gpu, DistributedDataParallel training.**
+
+```
+python -m torch.distributed.launch --nproc_per_node=8 \ # specify gpu number
+--master_port=29500  \
+train.py  --launcher pytorch \
+--config /path/to/config_file 
+```
+
+- specify the ```data_root``` in the config file;
+- log dir will be created in ```./work_dirs```;
+- download the [DeiT pretrained model](https://dl.fbaipublicfiles.com/deit/deit_base_distilled_patch16_384-d0272ac0.pth) and specify the ```pretrained``` path in the config file.
+
+
+<!-- **3. I try to take 'H/8*W/8 x 256' as the vector size for decoder input and remain stride 32 for the backbone.**
```

Comment:
The following content seems to be a conflict during git merge. Please confirm and delete unnecessary content.